parl/remote/remote_decorator.py

```diff
@@ -301,6 +301,22 @@ def __setattr__(self, attr, value):
         return proxy_wrapper
 
     max_memory = kwargs.get('max_memory')
-    if len(args) == 1 and callable(args[0]):
+    if len(args) == 1 and callable(args[0]):  # args[0]: cls
```

Comment:
Could you explain why we need to check these conditions and add comment for it?

Reply:
fixed